src/components/appLayout/OverviewLayout.tsx

```diff
@@ -0,0 +1,48 @@
+import * as React from 'react';
+import { Layout, theme } from 'antd';
+import { ReactNode } from 'react';
+
+const { Header, Sider, Content } = Layout;
+
+type OverviewLayoutProps = {
+  menu?: ReactNode;
+  headerContent?: ReactNode;
+  content?: ReactNode;
+  isMenuFolded: boolean;
+};
+
+const OverviewLayout: React.FC<OverviewLayoutProps> = (props: OverviewLayoutProps) => {
+  const {
+    token: { colorBgContainer }
+  } = theme.useToken();
+
+  const layoutStyle = {
+    height: '100vh'
+  };
+
+  const contentStyle = {
+    margin: '24px 16px',
+    padding: 24
+  };
+
+  const headerStyle = {
+    padding: 0,
+    background: colorBgContainer
+  };
+
+  return (
+    <Layout style={layoutStyle}>
+      <Sider trigger={null} collapsible collapsed={props.isMenuFolded}>
+        {props.menu}
+      </Sider>
+
+      <Layout className='site-layout'>
+        <Header style={headerStyle}>{props.headerContent}</Header>
+
+        <Content style={contentStyle}>{props.content}</Content>
+      </Layout>
+    </Layout>
+  );
+};
+
+export default OverviewLayout;
```

Comment:
i do not like default exports at all but this is a preference we can talk about some other time

Reply:
Switched to named imports for now. Need to look up the difference but we can talk about this. Such changes can be added to improvements later if needed.